### PR TITLE
[onert] add transposed flag into Conv kernel

### PR DESCRIPTION
**Pull request by dayo09**
- add explicit transposed flag
- not to use _kernel->data if the data is transposed and freed
- to solve assert in #4319

ONE-DCO-1.0-Signed-off-by: dayo09 <dayg502@gmail.com>

### CODE REVIEW

File: runtime/onert/backend/cpu/ops/ConvolutionLayer.cc [hunk 1]
```diff
@@ -57,10 +57,16 @@ void ConvolutionLayer::convFloat32()
   op_params.float_activation_max = output_activation_max;
 
   nnfw::cker::Conv &kernel = *_conv_kernel;
+
+  const float *kernel_data =
+      (kernel.is_transposed() == true)
+          ? nullptr // _kernel->buffer() is transformed into cker::Conv::_modified_filter_data
```
Comment (ragmani) [inline runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: Is it Okay if kernel_data is nullptr?

Comment (dayo09) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: Actually inside the cker::Conv kernel, if it expects transposed weights, it doesn't uses `kernel_data`(whatever comes as parameter).
Instead, it uses _modified_filter_data.

Comment (dayo09) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: And `bool usableMultiThreaded(...)` seems to be working as a 'Do Transpose' flag.

Comment (ragmani) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: Can I get why did you add this variable even though cker::Conv kernel doesn't use `kernel_data`? If the `kernel_data` is not used,  the as-is code seems to be fine.

Comment (dayo09) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: I am working on early-releasing tensor data- releases buffer when tensor's num_reference is zero.
I tried to put assert on buffer() when num_reference is zero.
And this arouse the assert.
I know this is so subtle change... But _transpose, for example, could be fine to use to clarify the meaning here.

Comment (ragmani) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: How about adding an assert for it instead of using `kernel_data`?

Comment (ragmani) [reply runtime/onert/backend/cpu/ops/ConvolutionLayer.cc:63]: I talked about this with @dayo09 offline. This line is needed if an assertion is added in `Tensor::buffer()` to check that the tensor's buffer is not nullptr. So we agree to add the code, @dayo09 is going to do it.

**Comment by dayo09** [comment]
### Issue : Assert on Reshape's negative test case

```
[2020-09-21T01:56:00.205Z] [ RUN      ] GenModelTest.neg_reshape_multiple_executions
[2020-09-21T01:56:00.205Z] Error during nnfw_session::run : Reshape: 2nd param is not compatible with the shape of input
[2020-09-21T01:56:00.205Z] [       OK ] GenModelTest.neg_reshape_multiple_executions (3 ms)
[2020-09-21T01:56:00.205Z] [ RUN      ] GenModelTest.OneOp_If
[2020-09-21T01:56:00.205Z] nnfw_api_gtest: /opt/jenkins_slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline/runtime/onert/core/include/backend/cpu_common/Tensor.h:91: virtual uint8_t* onert::backend::cpu_common::Tensor::buffer() const: Assertion `_num_references > 0' failed.
[2020-09-21T01:56:00.205Z] /jenkins/slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline/Product/out/test/command/unittest: line 80: 10184 Aborted                 $TEST_BIN $(get_gtest_option)
[2020-09-21T01:56:00.205Z] Product/out/unittest_standalone/nnfw_api_gtest failed... return code: 134
[2020-09-21T01:56:00.205Z] ============================================
[2020-09-21T01:56:00.205Z] Finishing set 3: Product/out/unittest_standalone/nnfw_api_gtest...
```

```c++
//DynamicShapeInference.cc
    if (output_shape != output->getShape() || output->buffer() == nullptr)
```
- output->buffer()==nullptr is being checked here

**Comment by ragmani** [comment]
```bash
15:58:40  [ RUN      ] GeneratedTests.abs_dynamic_nnfw
15:58:40  nnapi_gtest: /opt/jenkins_slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline/runtime/onert/core/include/backend/cpu_common/Tensor.h:94: virtual uint8_t* onert::backend::cpu_common::Tensor::buffer() const: Assertion `(_buffer == nullptr && _num_references == 0) || (_buffer != nullptr && _num_references > 0)' failed.
15:58:40  /jenkins/slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline/Product/out/test/command/unittest: line 80: 21981 Aborted                 $TEST_BIN $(get_gtest_option)
15:58:40  /jenkins/slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline/Product/out/unittest/nnapi_gtest failed... return code: 134
```

I think `(_buffer == nullptr && _num_references == 0) || (_buffer != nullptr && _num_references > 0)`  is logically correct. But there was an error by it. It seems to be a bug. 

@dayo09 
Could you remove the assertion from this PR and deal with the assertion in other issue or PR?

**Comment by dayo09** [comment]
Closing this because it is to add assert, but causing to many changes in other files.

**Pull request closed without merging**